backend/experiment/actions/base_action.py

```diff
@@ -1,9 +1,16 @@
+from .frontend_style import FrontendStyle
+
 class BaseAction(object):
     ID = 'BASE'
+    style = FrontendStyle()
 
-    def __init__(self):
+    def __init__(self, style = FrontendStyle()):
+        self.style = style
         pass
 
     def action(self):
-        self.view = self.ID
-        return self.__dict__
+        action_dict = self.__dict__
+        action_dict['view'] = self.ID
+        action_dict['style'] = self.style.to_dict()
+
+        return action_dict
```

Comment:
This could become problematic, as subclasses of `BaseAction` may implement their own `action` function.  The same goes for `__init__` - `super` is not always called by the subclasses.

Reply:
Hmm, what would be a good solution to this then? Ideally I would like to have all actions serialise their style so that the frontend knows what to expect. Is there another way to enforce that?

Reply:
You'd probably need to call `super()` either on the `__init__` (and then do `style.to_dict()` in the init function) or the `action` function of the subclasses to be sure.

Reply:
I think adding the `style.to_dict()` method in a custom implementation of the `action` method is the way to go. If it's ommitted, it will just mean no styles will be sent to the frontend, right? That shouldn't normally be a problem since every component has a (or not yet) default style.